pkg/vdbgen/vdb.go

```diff
@@ -475,6 +551,37 @@ func (d *DBGenerator) setSubclusterDetail(ctx context.Context) error {
 	return nil
 }
 
+// setImage will fetch the server version and use it to build
+// and set the image inside v.vdb.
+func (d *DBGenerator) setImage(ctx context.Context) error {
+	q := Queries[VersionQueryKey]
+	rows, err := d.Conn.QueryContext(ctx, q)
+	if err != nil {
+		return fmt.Errorf("failed running '%s': %w", q, err)
+	}
+	defer rows.Close()
+
+	if rows.Err() != nil {
+		return fmt.Errorf("failed running '%s': %w", q, rows.Err())
+	}
+	if !rows.Next() {
+		return errors.New("could not get Vertica version from meta-function")
+	}
+	var image string
+	if err := rows.Scan(&image); err != nil {
+		return fmt.Errorf("failed running '%s': %w", q, err)
+	}
+	// regex to match Vertica version
+	exp := regexp.MustCompile(`\d+(?:\.\d+){2}`)
+	version := exp.FindString(image)
+	if version == "" {
+		return errors.New("could not find Vertica version")
+	}
+	d.Objs.Vdb.Spec.Image = fmt.Sprintf("vertica/vertica-k8s:%s-0", version)
```

Comment:
I suggest we add some explanation about how we come up with the image name.
```suggestion
        // Pick an image we hosted in Docker Hub.  We always publish an image for
        // hotfix 0. Rarely do we publish one for subsequent hotfixes, so always use
        // hotfix 0 regardless of what hotfix was currently in use.
	d.Objs.Vdb.Spec.Image = fmt.Sprintf("vertica/vertica-k8s:%s-0", version)
```